raft/raft.go

```diff
@@ -172,6 +173,9 @@ func newRaft(id uint64, peers []uint64, election, heartbeat int, storage Storage
 	if !isHardStateEqual(hs, emptyState) {
 		r.loadState(hs)
 	}
+	if applied > 0 {
```

Comment:
Correct. Raft will panic if you pass an applied index that is less than firstIndex (or greater than HardState.Commit)